commands/channel_test.go

```diff
@@ -2176,3 +2166,243 @@ func (s *MmctlUnitTestSuite) TestRenameChannelCmd() {
+			EXPECT().
+			GetTeam(teamName, "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetTeamByName(teamName, "").
+			Return(foundTeam, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			CreateChannel(foundChannel).
+			Return(foundChannel, &model.Response{Error: nil}).
+			Times(1)
+
+		err := createChannelCmdF(s.client, cmd, args)
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 1)
+		s.Require().Equal(printer.GetLines()[0], foundChannel)
+	})
+
+	s.Run("should create private channel", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+
+		teamName := "teamName"
+		channelName := "channelName"
+		channelDisplayName := "channelDisplayName"
+		argsTeamChannel := teamName + ":" + channelName + ":" + channelDisplayName
+		args := []string{argsTeamChannel}
+
+		cmd.Flags().String("team", teamName, "Team Name")
+		cmd.Flags().String("name", channelName, "Channel Name")
+		cmd.Flags().String("display_name", channelDisplayName, "Channel Display Name")
+		cmd.Flags().Bool("private", true, "Create a private channel")
+
+		foundTeam := &model.Team{
+			Id:          "teamId",
+			Name:        teamName,
+			DisplayName: "teamDisplayName",
+		}
+
+		foundChannel := &model.Channel{
+			TeamId:      "teamId",
+			Name:        channelName,
+			DisplayName: channelDisplayName,
+			Type:        model.CHANNEL_PRIVATE,
+		}
+
+		s.client.
+			EXPECT().
+			GetTeam(teamName, "").
+			Return(foundTeam, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			CreateChannel(foundChannel).
+			Return(foundChannel, &model.Response{Error: nil}).
+			Times(1)
+
+		err := createChannelCmdF(s.client, cmd, args)
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 1)
+		s.Require().Equal(printer.GetLines()[0], foundChannel)
+	})
+
+	s.Run("should create channel with header and purpose", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+
+		teamName := "teamName"
+		channelName := "channelName"
+		channelDisplayName := "channelDisplayName"
+		header := "someHeader"
+		purpose := "somePurpose"
+		argsTeamChannel := teamName + ":" + channelName + ":" + channelDisplayName
+		args := []string{argsTeamChannel}
+
+		cmd.Flags().String("team", teamName, "Team Name")
+		cmd.Flags().String("name", channelName, "Channel Name")
+		cmd.Flags().String("display_name", channelDisplayName, "Channel Display Name")
+		cmd.Flags().String("header", header, "Channel header")
+		cmd.Flags().String("purpose", purpose, "Channel purpose")
+		cmd.Flags().Bool("private", true, "Create a private channel")
+
+		foundTeam := &model.Team{
+			Id:          "teamId",
+			Name:        teamName,
+			DisplayName: "teamDisplayName",
+		}
+
+		foundChannel := &model.Channel{
+			TeamId:      "teamId",
+			Name:        channelName,
+			DisplayName: channelDisplayName,
+			Header:      header,
+			Purpose:     purpose,
+			Type:        model.CHANNEL_PRIVATE,
+			CreatorId:   "",
```

Comment:
nit: this is already the default value of the field, so we don't need to specify it